src/app/public/modules/autonumeric/autonumeric.directive.ts

```diff
@@ -69,15 +70,28 @@ export class SkyAutonumericDirective implements OnInit, OnDestroy, ControlValueA
 
   /**
    * Assigns the name of a property from `SkyAutonumericOptionsProvider`.
+   * Exclusive to the IsoCurrencyCodeAndLocale Input.
    */
   @Input()
   public set skyAutonumeric(value: SkyAutonumericOptions) {
-    this.autonumericOptions = this.mergeOptions(value);
+    this.autonumericOptions = this.autonumericConfigService.getAutonumericOptions(value);
     this.updateAutonumericInstance();
   }
 
+  /**
+   * Formats a Currency based off a given ISO Currency Code and Locale.
+   * Exclusive to the skyAutonumeric Input.
+   */
+  @Input()
+  public set isoCurrencyCodeAndLocale(value: { isoCurrencyCode: string, locale?: string }) {
```

Comment:
Talked to the team and we think the best way to handle this is to create a new input named `skyAutonumericDefaults`, which would have a subproperty of `currency`. This way, the input can be extensible and provide more configuration options in the future (for example, percentages based on locales).
```
<input
  type="text"
  [skyAutonumeric]="{} || 'NorthAmerican'"
  [skyAutonumericDefaults]="{
    currency: {
      isoCurrencyCode: '',
      locale: 'en-CA'
    },
    // Just by way of example:
    percent: {
      locale: 'en-US'
    }
  }"
/>
```

Reply:
This is a lot better so I removed the `isoCurrencyCodeAndLocale` input and replaced it with your suggestion.

I modelled the Autonumeric Options as:
1. Start with empty options.
2. Apply global options from `SkyAutonumericOptionsProvider`.
3. Apply options passed into `skyAutonumeric` input.
4. If `skyAutonumericDefaults` has a defined Sky Default get the options and apply them. 

That means that any defined options from the `skyAutonumeric` input will be overwritten by `skyAutonumericDefaults`. I could see the precedence order of those two inputs being switched.